recipes/repo.rb

```diff
@@ -374,415 +374,70 @@
   end
 end
 
-# Pulp setup
-pulp_data_directory = '/var/repos/.pulp'
-user 'pulp' do
-  manage_home true
-  system true
-  shell '/usr/sbin/nologin'
-  comment 'Pulp content manager'
-  uid '1200'
-end
-group 'docker' do
-  append true
-  members ['pulp']
-  action [:manage]
-end
-group 'gpg-vault' do
-  append true
-  members ['pulp']
-  action [:manage]
-end
 
-execute 'gpg-init' do
-  command 'gpg -K'
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  creates '/home/pulp/.gnupg'
-end
-cookbook_file '/home/pulp/.gnupg/gpg.conf' do
-  source 'gpg-vault.conf'
-  owner 'pulp'
-  group 'pulp'
-  mode '0600'
-end
-execute "gpg --import /var/repos/repos.key" do
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  not_if "gpg --list-keys #{gpg_key['fingerprint']}"
-end
-execute 'trust_public_key' do
-  command "(echo 5; echo y; echo save) | gpg --command-fd 0 --no-tty --no-greeting -q --edit-key #{gpg_key['fingerprint']} trust"
-  user 'pulp'
-  group 'pulp'
-  environment 'HOME' => '/home/pulp'
-  not_if "gpg --list-keys #{gpg_key['fingerprint']} | grep -q '\\[ultimate\\]'"
+# TODO: Start of pulp cleanup to remove in next major release.
+node['ros_buildfarm']['rpm_repos'].keys.each do |dist|
+  dist_dir = "/var/repos/#{dist}_pulp"
+  directory dist_dir do
+    action :delete
+    recursive true
+  end
 end
 
-directory pulp_data_directory do
-  owner 'pulp'
-  mode '0700'
+0.upto(10) do |i|
+  service "pulp-worker@#{i}" do
+    action [:stop, :disable]
+    only_if { File.exist?("/etc/systemd/system/pulp-worker@.service") }
+  end
 end
-directory "#{pulp_data_directory}/media" do
-  owner 'pulp'
+file "/etc/systemd/system/pulp-worker@.service" do
+  action :delete
 end
-directory "#{pulp_data_directory}/docker_image" do
-  owner 'pulp'
+%w(pulp-api-endpoint pulp-content-endpoint pulp-resource-manager pulp-rsync-endpoint).each do |service_name|
+  service service_name do
+    action [:stop, :disable]
+    only_if { File.exist?("/etc/systemd/system/#{service_name}.service") }
+  end
+  file "/etc/systemd/system/#{service_name}.service" do
+    action :delete
+  end
 end
 
-cookbook_file "#{pulp_data_directory}/initialize.py" do
-  owner 'pulp'
-  source 'pulp/rpm_repo_init.py'
-  mode '0600'
+file '/usr/local/bin/systemd-docker' do
+  action :delete
 end
 
-package 'redis'
-service 'redis' do
-  action [:start, :enable]
+service 'postgresql' do
+  action [:stop, :disable]
+  only_if { File.exist?('/usr/lib/systemd/system/postgresql.service') }
```

Comment:
Double checking: I can not find this path on my focal system. Was it produced by a custom previous installation?, should it `/lib/systemd/system/postgresql.service`?

Same for redis below.

Reply:
> Double checking: I can not find this path on my focal system. Was it produced by a custom previous installation?, should it /lib/systemd/system/postgresql.service?

In my test systems `/usr/lib` and `/lib` are one and the same. But the reported value in `systemctl status postgresql.service` canonicalizes "bare" `/lib` so I'll do likewise.

Reply:
Done in 282425188618b1619de5d85871985f072e76abf8